lib/IRGen/Linking.cpp

```diff
@@ -317,8 +329,10 @@ std::string LinkEntity::mangleAsString() const {
     assert(isa<AbstractFunctionDecl>(getDecl()));
     std::string Result;
     if (auto *Constructor = dyn_cast<ConstructorDecl>(getDecl())) {
-      Result = mangler.mangleConstructorEntity(Constructor, true,
-                                               /*isCurried=*/false);
+      bool useAllocatorMangling = shouldUseAllocatorMangling(Constructor);
```

Comment:
I would rather we had two separate LinkEntity kinds, DynamicallyReplaceableFunctionImpl and DynamicallyReplaceableAllocatorImpl. We already do this for method descriptors and dispatch thunks. Alternatively we could have a separate bit in LinkEntity for this purpose.